### PR TITLE
Ajax filter by labels

### PR DESCRIPTION
**Pull request by volkov97**
Filter by labels with ajax.

### CODE REVIEW

File: common.blocks/labels-list/__label/labels-list__label.deps.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
 ({
-    shouldDeps: 'link'
+    mustDeps: 'i-bem-dom',
+    shouldDeps: [
+        'link',
+        'location',
+        'link'
```
Comment (LISBON11) [inline common.blocks/labels-list/__label/labels-list__label.deps.js]: 2 раза link?

File: common.blocks/labels-list/__label/labels-list__label.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            var queryParams = Location.getUri().queryParams;
+            var labels = (queryParams.labels && queryParams.labels[0]) ? queryParams.labels[0].split(',') : [];
+            var labelTitle = this._link.params.label;
+
+            if (this._link.hasMod('active')) {
+                this._link.delMod('active');
+                this._link.domElem[0].style['background-color'] = this._link.params.usualColor;
+
+                // if this label is active then it should be in this array
+                labels.splice(labels.indexOf(labelTitle), 1);
```
Comment (LISBON11) [inline common.blocks/labels-list/__label/labels-list__label.js]: + Если тебе приедет страница  /?labels=bem-components  при перезагрузке, а потом ты кликнешь по  лэйблу bem-components то будет что-то вроде /?labels=bem-components%2Cbem-components и неверная поисковая выдача.  
+  По идее тебе надо при перезагрузке  страницы нужно смотреть queryparams  и подсвечивать в сайде нужные теги, которые уже есть в урле . и ставить им модификатор active_true  и тогде они не будут пушиться эгэйн

File: common.blocks/labels-list/__label/labels-list__label.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            } else {
+                this._link.setMod('active');
+                this._link.domElem[0].style['background-color'] = this._link.params.activeColor;
+
+                labels.push(labelTitle);
+            }
```
Comment (LISBON11) [inline common.blocks/labels-list/__label/labels-list__label.js]: + может стоит сделать как то так? тоглить модификатор а его колбеке уже менять цвет
```js
onSetMod: {
    js: {
        inited: function() {
            this._link = this.findChildBlock(Link);
        }
    },

    active: function(elem, modVal) {
        this._link.domElem[0].style['background-color'] =
            this._link.params[modVal === 'true' ? 'activeColor' :  'usualColor'];
    }

},


this.hasMod('active') ?
     labels.splice(labels.indexOf(labelTitle), 1) :
     labels.push(labelTitle);

this.toggleMod('active');
```

File: common.blocks/labels-list/__label/labels-list__label.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    ['i-bem-dom', 'location', 'link'],
+    function(provide, bemDom, Location, Link) {
+
+    var Label = bemDom.declElem('labels-list', 'label', {
+        onSetMod: {
+            js: {
+                inited: function() {
+                    this._link = this.findChildBlock(Link);
+
+                    this._events(this._link).on('click', this._onLabelClick);
```
Comment (tadatuta) [inline common.blocks/labels-list/__label/labels-list__label.js]: нужно сделать ленивую инициализацию

File: common.blocks/labels-list/__label/labels-list__label.deps.js (hunk 1 above)
Comment (volkov97) [reply common.blocks/labels-list/__label/labels-list__label.deps.js]: упс)

File: common.blocks/labels-list/__label/labels-list__label.js (hunk 2 above)
Comment (volkov97) [reply common.blocks/labels-list/__label/labels-list__label.js]: Изменил подход к этому заданию - логика похожая стала, да

File: common.blocks/labels-list/__label/labels-list__label.js (hunk 3 above)
Comment (volkov97) [reply common.blocks/labels-list/__label/labels-list__label.js]: да, так выглядит лучше

File: common.blocks/labels-list/__label/labels-list__label.js (hunk 4 above)
Comment (volkov97) [reply common.blocks/labels-list/__label/labels-list__label.js]: сделал

File: common.blocks/labels-list/__label/labels-list__label.bemtree.js [hunk 5]
```diff
@@ -15,10 +15,15 @@ block('labels-list').elem('label').content()(function() {
         url: '/?labels=' + name,
         attrs: {
             style: [
-                `background-color: ${calcBrightness(color, 80)}`,
+                `background-color: ${calcBrightness(color, 95)}`,
```
Comment (tadatuta) [inline common.blocks/labels-list/__label/labels-list__label.bemtree.js]: let's move 95 and 40 to constants at the top of the file

File: common.blocks/labels-list/__label/labels-list__label.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            var queryLabels = Location.getUri().queryParams.labels;
+            var activeLabels = (queryLabels && queryLabels[0]) ? queryLabels[0].split(',') : [];
+
+            var labelName = this.params.name;
+
+            activeLabels.indexOf(labelName) === -1 ?
+                activeLabels.push(labelName) :
+                activeLabels.splice(activeLabels.indexOf(labelName), 1);
+
+            Location.change(
```
Comment (tadatuta) [inline common.blocks/labels-list/__label/labels-list__label.js:29]: should drop pagination

Comment (volkov97) [reply common.blocks/labels-list/__label/labels-list__label.js:29]: dropped

**Review by tadatuta — APPROVED** [review]
